src/averageAges.js

```diff
@@ -15,7 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
+  let totalAge = 0;
+  let count = 0;
+
+  people.forEach(person => {
+    if (person.sex === 'm' && (!century
+      || Math.ceil(person.died / 100) === century)) {
+      totalAge += person.died - person.born;
+      count++;
+    }
+  });
+
+  return count === 0 ? 0 : totalAge / count;
```

Comment:
Instead of checking if count equals 0 at the end, check at the start if the array is empty and if it is, return 0.